dateparser/date.py

```diff
@@ -457,7 +457,7 @@ def date_strings():
                 yield stripped_date_string
 
         if self.try_previous_locales:
-            for locale in self.previous_locales:
+            for locale in self.previous_locales.keys():
```

Comment:
:lipstick: 

```suggestion
            for locale in self.previous_locales:
```

I’m just mentioning this in case `.keys()` was not added on purpose. But given `.keys()` is more explicit, maybe the current code is more readable. :heavy_check_mark: either way.

Reply:
I think we can leave it as-is. For me, it seems more accurate to explicitly indicate that we iterate through the keys. :+1: